coops-ui/src/views/Dashboard.vue

```diff
@@ -93,7 +104,82 @@ export default {
     goToStandaloneAddresses () {
       this.setCurrentFilingStatus('NEW')
       this.$router.push({ name: 'standalone-addresses', params: { id: 0 } }) // 0 means "new COA filing"
+    },
+
+    checkToReloadDashboard () {
+      // cancel any existing timer so we can start fresh here
+      clearTimeout(this.refreshTimer)
+
+      let filingId = null
+      if (this.$route !== undefined) filingId = +this.$route.query.filing_id // if missing, this is NaN
+
+      // only consider refreshing the dashboard if we came from a filing
+      if (!filingId) return
+
+      let isInFilingHistory = this.historyFilings.filter(el => el.filingId === filingId).length > 0
```

Comment:
Trivial, but using Array.prototype.find() might be a bit faster since it should return when a match is found instead of processing all elements.